### ISSUE TITLE
Background mode (wakelock?) in the app

### ISSUE THREAD

**Issue by vit1-irk** [Staacks/phyphox-android#5]
Please add the optional feature of continuation of the experiment in the background mode.

When making a long-time measurement it can happen that the device is locked (or the screen turned off, or you switch to another app, whatever), and app pauses the experiment. Sometimes it is a trouble to gather the data in such a case.

// BTW, congratulations with F-Droid release, phyphox is great to appear here

**Comment by Staacks** [comment]
Thanks, this is on our todo list, but only as a long-term goal without a specific time at which we plan to start on this task.
The problem here is that the app needs to be split into the user interface part and the data acquisition and analysis part, so the former can run as an activity (Android terminology for a process with visible user interface) and the latter as a service (a process running in the background, typically showing an icon in the notification bar). At the moment, phyphox is an activity (technically speaking it consists of multiple activities) and activities are stopped by the OS as soon as they stop being visible and there is no way around that. Splitting however should be possible (not sure what is actually allowed for a process and what not), but it requires some major restructuring of the app, which is why we have not tackled it yet.

**Comment by vit1-irk** [comment]
Also be careful about services.
In the latest Android versions (Oreo and Pie) background services are restricted in the execution time, so it is needed to create a "persistent notification" during the measurement to ensure the service will not stop.

**Comment by nakic** [comment]
I was about to use phyphox to collect pavement smoothness data while cycling around collecting street view imagery. I need to have a map on screen to do something like that, which unfortunately kills phyphox, so I have to do without it or find something else that can get it done. Just my 2c.

**Comment by Staacks** [comment]
Please, let us know if you find something. Would be encouraging to see what is actually possible in the background.

**Comment by nakic** [comment]
I created a feature request to record accellerometer data in OSMAnd: it seemed a much easier task there than it is here, given the apps' architectures. We'll see what happens.

**Comment by Staacks** [comment]
OSMAnd is the map app in question and you asked them to log sensor data? Seems indeed like the easiest solution if they do that, but it's not the reference case I would have hoped for as their app would be in the foreground. Still, for your case a simple solution, indeed.

**Comment by nakic** [comment]
Yup, just trying to be pragmatic. FWIW, there are action cameras (e.g Garmin Virb) which have location and acceleration sensors and produce GPX output with accelerometer data out of the box, so I'd imagine there's already a need for something like that (as marginal as it might be).

It's a drop in the ocean in terms of the kinds of sensing phyphox supports, of course, but given that no progress on background execution was made in 3 years, it seemed reasonable to try the other way. If it's not a priority, it's not a priority: that's fine, but I'm doing what I can to find a solution to the immediate problem.

**Comment by Noitarud** [comment]
Not sure if limited to tablets,but it allows two up in split screen mode, or more with freeform (but one media player can play...).

**Comment by lgommans** [comment]
For reference, the OsmAnd issue is still open and labeled as 'nice-to-have' with no further activity since. https://github.com/osmandapp/OsmAnd/issues/12227
